urnai/sc2/environments/sc2environment.py

```diff
@@ -0,0 +1,115 @@
+from pysc2.lib import actions, features
+
+from urnai.environments.environment_base import EnvironmentBase
+
+
+class SC2Env(EnvironmentBase):
+    def __init__(
+            self,
+            map_name='Simple64',
+            player_race=sc2_env.Race.random,
+            enemy_race=sc2_env.Race.random,
+            difficulty=sc2_env.Difficulty.very_easy,
+            visualize=False,
+            reset_done=True,
+            spatial_dim=16,
+            step_mul=16,
+            game_steps_per_episode=0,
+            realtime=False,
+            self_play=False,
+            action_space=actions.ActionSpace.RAW,
+            use_raw_units=True,
+            raw_resolution=64,
+            use_feature_units=True,
+            screen=64,
+            minimap=64,
+    ):
+        super().__init__(map_name, visualize, reset_done)
+
+        FLAGS = flags.FLAGS
+        FLAGS(sys.argv)
+
+        self.self_play = self_play
+        self.step_mul = step_mul
+        self.game_steps_per_episode = game_steps_per_episode
+        self.spatial_dim = spatial_dim
+        self.player_race = player_race
+        self.enemy_race = enemy_race
+        self.difficulty = difficulty
+        if self.self_play:
+            self.players = [
+                sc2_env.Agent(self.player_race),
+                sc2_env.Agent(self.enemy_race),
+            ]
+        else:
+            self.players = [
+                sc2_env.Agent(self.player_race),
+                sc2_env.Bot(self.enemy_race, self.difficulty),
+            ]
+        self.realtime = realtime
+        self.done = False
+
+        self.start(action_space, use_raw_units, raw_resolution, 
+                   use_feature_units, screen, minimap)
+
+    def start(
+            self,
+            action_space=actions.ActionSpace.RAW,
+            use_raw_units=True,
+            raw_resolution=64,
+            use_feature_units=True,
+            screen=64,
+            minimap=64,
+    ) -> None:
+        self.done = False
+        if self.env_instance is None:
+            self.env_instance = sc2_env.SC2Env(
+                map_name=self.map_name,
+                visualize=self.visualize,
+                players=self.players,
+                agent_interface_format=features.AgentInterfaceFormat(
+                    action_space=action_space,
+                    use_raw_units=use_raw_units,
+                    raw_resolution=raw_resolution,
+                    use_feature_units=use_feature_units,
+                    feature_dimensions=features.Dimensions(screen, minimap),
+                ),
+                step_mul=self.step_mul,
+                game_steps_per_episode=self.game_steps_per_episode,
+                realtime=self.realtime,
+            )
+
+    def step(self, action: actions.FunctionCall) -> Tuple[List[Any], int, bool]:
+        timestep = self.env_instance.step(action)
+        obs, reward, done = self.parse_timestep(timestep)
```

Comment:
It was told in today's meeting that `done` needs to be changed to two different booleans: `terminated`, which determines if the objective was achieved, and `truncated`, which tells if the maximum number of steps was reached.

Reply:
I don't think these are booleans, but statuses. In this case it would be better to use `Enum`, `string` or `constants`.